cdap-features/src/main/java/io/cdap/cdap/features/Feature.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.features;
+
+import io.cdap.cdap.api.FeatureFlagsProvider;
+import io.cdap.cdap.api.PlatformInfo;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Defines Features Flags to be used in CDAP.
+ * Features take the version that they were introduced as a first parameter. Optionally they can take a
+ * second paramenter to define their default behavior if they are not present in configuration.
+ * By default features default to enabled after they are introduced, and disabled before they were introduced
+ */
+public enum Feature {
+  ALLOW_FEATURE_FLAGS("6.6.0");
+  
+  public static final String FEATURE_FLAG_PREFIX = "feature.";
```

Comment:
I'd move this to `FeatureFlagsProvider` and `extractFeatureFlags` to some utility class. This way we won't need a dependency on features module by the platform. Basically, Feature enum will become a library only for feature flag user